src/lib/device-types/index.ts

```diff
@@ -80,15 +81,23 @@ const getFirstValidBuild = (
 		return Promise.resolve() as Promise<BuildInfo | undefined>;
 	}
 
-	return getBuildData(slug, versions[0]).then(buildInfo => {
-		if (!buildInfo.ignored && buildInfo.deviceType) {
-			// TS can't infer this correctly and gets confused when
-			// checking it against the Promise return value
-			return buildInfo as BuildInfo;
-		}
+	const buildId = versions[0];
+	return getBuildData(slug, buildId)
+		.catch(err => {
+			captureException(
+				err,
+				`Failed to get device type build data for ${slug}/${buildId}`,
+			);
+		})
+		.then(buildInfo => {
+			if (buildInfo && !buildInfo.ignored && buildInfo.deviceType) {
```

Comment:
Yea but we need it this way so that we can reach to the recursive `getFirstValidBuild()` when we fail to get a build or is it doesn't have a device-type.json or it's ignored.
Otherwise we would have to also have the `getFirstValidBuild()` call inside the `.catch()`. Except if you have something different in mind?

Reply:
Ah, that's a bit awkward but makes sense